### PR TITLE
fp16_speedup

### PR DESCRIPTION
**Pull request by rydenisbak**
PR from this issue
https://github.com/open-mmlab/mmdetection/issues/2131

### CODE REVIEW

File: mmdet/core/fp16/hooks.py [hunk 1]
```diff
@@ -97,8 +97,12 @@ def wrap_fp16_model(model):
 def patch_norm_fp32(module):
     if isinstance(module, (nn.modules.batchnorm._BatchNorm, nn.GroupNorm)):
         module.float()
-        module.forward = patch_forward_method(module.forward, torch.half,
-                                              torch.float)
+        path_forward_condition = isinstance(module, nn.GroupNorm)
+        major, minor = [int(v) for v in torch.__version__.split('.')][:2]
+        path_forward_condition |= major < 1 or major == 1 and minor < 3
+        if path_forward_condition:
```
Comment (hellock) [inline mmdet/core/fp16/hooks.py]: The condition can be rewritten in just one line:
```python
if torch.__version__ < (1, 3) or isinstance(module, nn.GroupNorm):
    module.forward = patch_forward_method(module.forward, torch.half, torch.float)
```

Comment (rydenisbak) [reply mmdet/core/fp16/hooks.py]: image
but I can write
image
It looks more pretty, thanks

**Comment by hellock** [comment]
Thanks! Could you fix the linting error?

**Comment by rydenisbak** [comment]
@hellock I found some bug in the code.
My torch version is '1.3.1+cu100' and it cause problem in line
https://github.com/rydenisbak/mmdetection/blob/fp16_speedup/mmdet/core/fp16/hooks.py#L100
```python
torch_version = [int(v) for v in torch.__version__.split('.')]
```
We need to change it, for example like this
```python
torch_version = [int(v) for v in torch.__version__.split('.') if v.isdigit()]
```
or like this
```python
torch_version = [int(v) if v.isdigit() else int(v.split('+')[0]) for v in torch.__version__.split('.') if v.isdigit()]
```
or like something else

**Comment by hellock** [comment]
Oh the pytorch versions installed by pip and conda are different. Maybe simply
```python
torch_version = [int(v) for v in torch.__version__.split('.')[:2]]
```

**Comment by rydenisbak** [comment]
ok, good

**Pull request merged**